tools/flaky-test-reporter/slack_notification.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// slack_notification.go sends notifications to slack channels
+
+package main
+
+import (
+	"io/ioutil"
+	"strings"
+	"time"
+	"log"
+	"fmt"
+	"encoding/json"
+
+	"net/http"
+	"net/url"
+
+	"github.com/knative/test-infra/shared/testgrid"
+)
+
+const (
+	slackChatPostMessageURL = "https://slack.com/api/chat.postMessage"
+	// default filter for testgrid link
+	testgridFilter          = "exclude-non-failed-tests=20"
+)
+
+var (
+	slackUsername = "Knative bot"
+	slackTokenStr string
```

Comment:
these two can be part of the slack client

Reply:
moved